browser(webkit): postpone creation of the first page

dgozman opened the pull request:
When we create the first page in the default context in headless mode on mac,
it gets `NSWindow` that is "not visible". Although we call `[window setIsVisible:YES]`,
later on `window.isVisible` still returns `NO`.

We create our offscreen "headless" `NSWindow` directly from `applicationDidFinishLaunching:`.
Experiments show that delaying this by 100ms makes everything work. As a symptom,
we get `applicationDidUnhide:` notification that does not happen when we create the window
immediately.

Perhaps, we create the window too early, and there is some essential initialization
that happens after `applicationDidFinishLaunching:`. However, if we call
`[NSApp activateIgnoringOtherApps:YES]` like we do in headful mode, everything works.

The only solution that worked so far is creating the first page after a timeout.

yury-s commented on browser_patches/webkit/embedder/Playwright/mac/AppDelegate.m:
```diff
@@ -250,8 +250,18 @@ - (void)applicationDidFinishLaunching:(NSNotification *)aNotification
+                                    repeats: NO
+                                      block:(void *)^(NSTimer* timer)
+    {
+        [self createNewPage:0 withURL:_initialURL ? _initialURL : @"about:blank"];
```
Would it make sense to simply create the page on the client side (and always start in _noStartupWindow mode) ?

dgozman committed:
browser(webkit): postpone creation of the first page

When we create the first page in the default context in headless mode on mac,
it gets NSWindow that is "not visible". Although we call [window setIsVisible:YES],
later on window.isVisible still returns NO.

We create our offscreen "headless" NSWindow directly from applicationDidFinishLaunching:.
Experiments show that delaying this by 100ms makes everything work. As a symptom,
we get applicationDidUnhide: notification that does not happen when we create the window
immediately.

Perhaps, we create the window too early, and there is some essential initialization
that happens after applicationDidFinishLaunching:. However, if we call
[NSApp activateIgnoringOtherApps:YES] like we do in headful mode, everything works.

The only solution that worked so far is creating the first page after a timeout.
